File: tools/generate_customer_photos_manifest.py
#!/usr/bin/env python3

import json
import re
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
PHOTOS_DIR = ROOT / "img" / "customer_photos"
MANIFEST_PATH = PHOTOS_DIR / "manifest.json"
ALLOWED = {".jpg", ".jpeg", ".png", ".webp", ".svg"}


def sort_key(name: str):
    m = re.search(r"vinvel_cus_(\d+)", name)
    if m:
        return (0, int(m.group(1)))
    return (1, name.lower())


def main() -> int:
    if not PHOTOS_DIR.exists() or not PHOTOS_DIR.is_dir():
        raise SystemExit(f"Missing folder: {PHOTOS_DIR}")

    files = [
        p.name
        for p in PHOTOS_DIR.iterdir()
        if p.is_file() and p.suffix.lower() in ALLOWED and not p.name.startswith(".")
    ]
    files = sorted(files, key=sort_key)

    manifest = {
        "basePath": "img/customer_photos/",
        "files": files,
    }

    MANIFEST_PATH.write_text(json.dumps(manifest, indent=2) + "\n", encoding="utf-8")
    print(f"Wrote {MANIFEST_PATH.relative_to(ROOT)} with {len(files)} files")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: tools/generate_customer_photos_manifest.mjs
import {readdir, stat, writeFile} from 'node:fs/promises'
import path from 'node:path'

const ROOT = process.cwd()
const ALLOWED_EXT = new Set(['.jpg', '.jpeg', '.png', '.webp', '.svg'])

function sortKeyCustomerPhotos(name) {
  const m = /vinvel_cus_(\d+)/i.exec(name)
  if (m) return [0, Number(m[1])]
  return [1, name.toLowerCase()]
}

function sortKeyAlpha(name) {
  return [0, name.toLowerCase()]
}

async function buildManifest({dir, basePath, sortKeyFn}) {
  const entries = await readdir(dir)
  const files = []

  for (const name of entries) {
    if (name.startsWith('.')) continue
    const full = path.join(dir, name)
    const info = await stat(full)
    if (!info.isFile()) continue

    const ext = path.extname(name).toLowerCase()
    if (!ALLOWED_EXT.has(ext)) continue

    files.push(name)
  }

  files.sort((a, b) => {
    const [ka0, ka1] = sortKeyFn(a)
    const [kb0, kb1] = sortKeyFn(b)
    if (ka0 !== kb0) return ka0 - kb0
    if (typeof ka1 === 'number' && typeof kb1 === 'number') return ka1 - kb1
    return String(ka1).localeCompare(String(kb1))
  })

  return {
    basePath,
    files,
  }
}

async function writeManifest({dir, manifest}) {
  const out = path.join(dir, 'manifest.json')
  await writeFile(out, JSON.stringify(manifest, null, 2) + '\n', 'utf8')
  return out
}

async function main() {
  const customerDir = path.join(ROOT, 'img', 'customer_photos')
  const partnersDir = path.join(ROOT, 'img', 'partners')

  const customerManifest = await buildManifest({
    dir: customerDir,
    basePath: 'img/customer_photos/',
    sortKeyFn: sortKeyCustomerPhotos,
  })
  await writeManifest({dir: customerDir, manifest: customerManifest})
  console.log(`Wrote img/customer_photos/manifest.json with ${customerManifest.files.length} files`)

  const partnersManifest = await buildManifest({
    dir: partnersDir,
    basePath: 'img/partners/',
    sortKeyFn: sortKeyAlpha,
  })
  await writeManifest({dir: partnersDir, manifest: partnersManifest})
  console.log(`Wrote img/partners/manifest.json with ${partnersManifest.files.length} files`)
}

await main()
